SPECS/mdevctl/mdevctl.spec

```
@@ -0,0 +1,249 @@
+Source1:        https://github.com/mdevctl/mdevctl/releases/download/v%{version}/mdevctl-%{version}-vendor.tar.gz
+
+BuildRequires:  cargo
+BuildRequires:  make
+BuildRequires:  python3-docutils
+BuildRequires:  systemd-rpm-macros
+
+ExclusiveArch:  x86_64
+
+%ifarch x86_64
+%define rust_def_target x86_64-unknown-linux-gnu
+%define cargo_pkg_feature_opts --no-default-features
+%endif
+%ifarch aarch64
+%define rust_def_target aarch64-unknown-linux-gnu
+%define cargo_pkg_feature_opts --no-default-features
+%endif
+
+%global bashcompletiondir %{_datadir}/bash-completion/completions
+
+%description
+mdevctl is a utility for managing and persisting devices in the
+mediated device device framework of the Linux kernel.  Mediated
+devices are sub-devices of a parent device (ex. a vGPU) which
+can be dynamically created and potentially used by drivers like
+vfio-mdev for assignment to virtual machines.
+
+%prep
+%autosetup -n %{crate}-%{version} -p1
+
+# Do vendor expansion here manually by
+# calling `tar x` and setting up 
+# .cargo/config to use it.
+tar fx %{SOURCE1}
+mkdir -p .cargo
+
+cat >.cargo/config << EOF
+[source.crates-io]
+replace-with = "vendored-sources"
+
+[source.vendored-sources]
+directory = "vendor"
+EOF
+
+%build
+# Makefile.in is in the source tarball
+# but there is no configure.  Manually
+# build using `cargo build`
+cargo build --release --target=%{rust_def_target} %{cargo_pkg_feature_opts} --offline
+
+%install
+# Makefile.in is in the source tarball
+# but there is no configure.  Manually
+# install.  Exclude man8 files.
+mkdir -p %{buildroot}%{_sysconfdir}/mdevctl.d
+mkdir -p %{buildroot}%{_udevrulesdir}/
+install -m 644 60-mdevctl.rules %{buildroot}%{_udevrulesdir}/
+mkdir -p %{buildroot}%{_sbindir}
+install -m 755 ./target/%{rust_def_target}/release/mdevctl %{buildroot}%{_sbindir}
+ln -sf mdevctl %{buildroot}%{_sbindir}/lsmdev
+mkdir -p %{buildroot}%{bashcompletiondir}/
+install -m 644 -T $(find ./target/%{rust_def_target}/release -name mdevctl.bash) %{buildroot}%{bashcompletiondir}/mdevctl
+install -m 644 -T $(find ./target/%{rust_def_target}/release -name lsmdev.bash) %{buildroot}%{bashcompletiondir}/lsmdev
+mkdir -p %{buildroot}%{_libdir}/mdevctl/scripts.d/callouts
+mkdir -p %{buildroot}%{_libdir}/mdevctl/scripts.d/notifiers
+
+%if %{with check}
+%check
+# Use simple cargo test rather than cargo_test macro which seems to fail
+# because there is no rpm profile in th ecargo setup (seems like this file
+# needs cargo_prep to enable cargo_test macro)
+cargo test
+%endif
+
+%files
+%license COPYING
+%doc README.md
+%{_sbindir}/mdevctl
+%{_sbindir}/lsmdev
+%{_udevrulesdir}/60-mdevctl.rules
+%dir %{_sysconfdir}/mdevctl.d
+%dir %{_prefix}/lib/mdevctl/scripts.d/callouts
+%dir %{_prefix}/lib/mdevctl/scripts.d/notifiers
+# Exclude man8 files.
+%{_datadir}/bash-completion/completions/mdevctl
+%{_datadir}/bash-completion/completions/lsmdev
```

Since we're already defining the `bashcompletiondir` macro, let's use it in the `%files` section as well to avoid unncessary build breaks in case it's ever changed:
```suggestion
%{bashcompletiondir}/mdevctl
%{bashcompletiondir}/lsmdev
```